pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -198,42 +198,28 @@ func (p *Predictor) Reconcile(ctx context.Context, isvc *v1beta1.InferenceServic
 	// Handle InferenceService status updates based on the force stop annotation.
 	// If true, transition the service to a stopped and unready state; otherwise, ensure it's not marked as stopped.
 	if utils.GetForceStopRuntime(isvc) {
-		// Exit early if we have already set the status to stopped
-		existingStoppedCondition := isvc.Status.GetCondition(v1beta1.Stopped)
-		if existingStoppedCondition != nil && existingStoppedCondition.Status == corev1.ConditionTrue {
+		// Exit early if we have already set the predictor's status to stopped
+		existingPredictorCondition := isvc.Status.GetCondition(v1beta1.PredictorReady)
+		if existingPredictorCondition != nil && existingPredictorCondition.Status == corev1.ConditionFalse && existingPredictorCondition.Reason == v1beta1.StoppedISVCReason {
 			return ctrl.Result{}, nil
 		}
 
+		// Preserve the deployment mode value
 		deployMode := isvc.Status.DeploymentMode
 
 		// Clear all statuses
 		isvc.Status = v1beta1.InferenceServiceStatus{}
-
-		// Preserve the deployment mode value
 		isvc.Status.DeploymentMode = deployMode
 
-		// Set the ready condition
-		predictorReadyCondition := &apis.Condition{
+		// Set the predictor's ready condition
+		predictorStoppedCondition := &apis.Condition{
 			Type:   v1beta1.PredictorReady,
 			Status: corev1.ConditionFalse,
 			Reason: v1beta1.StoppedISVCReason,
 		}
-		isvc.Status.SetCondition(v1beta1.PredictorReady, predictorReadyCondition)
-
-		// Add the stopped condition
-		stoppedCondition := &apis.Condition{
-			Type:   v1beta1.Stopped,
-			Status: corev1.ConditionTrue,
-		}
-		isvc.Status.SetCondition(v1beta1.Stopped, stoppedCondition)
+		isvc.Status.SetCondition(v1beta1.PredictorReady, predictorStoppedCondition)
```

Comment:
What was the reason for setting `predictorStoppedCondition` under the `PredictorReady` condition? The state representation feels a bit confusing.

Reply:
the stopped status was moved to the controller.go so it is set after all the reconcilers are called